tests2/pgtests.py

```diff
@@ -536,24 +536,25 @@ def test_cursor_messages(self):
 
         # using INFO message level because they are always sent to the client regardless of
         # client_min_messages: https://www.postgresql.org/docs/11/runtime-config-client.html
-        self.cursor.execute("""
-            CREATE OR REPLACE PROCEDURE test_cursor_messages()
-            LANGUAGE plpgsql
-            AS $$
-            BEGIN
-                RAISE INFO 'hello world' USING ERRCODE = '01000';
-            END;
-            $$;
-        """)
-        self.cursor.execute("CALL test_cursor_messages();")
-        self.assertTrue(type(self.cursor.messages) is list)
-        self.assertEqual(len(self.cursor.messages), 1)
-        self.assertTrue(type(self.cursor.messages[0]) is tuple)
-        self.assertEqual(len(self.cursor.messages[0]), 2)
-        self.assertTrue(type(self.cursor.messages[0][0]) is unicode)
-        self.assertTrue(type(self.cursor.messages[0][1]) is unicode)
-        self.assertEqual('[01000] (-1)', self.cursor.messages[0][0])
-        self.assertTrue(self.cursor.messages[0][1].endswith('hello world'))
+        for message in ('hello world', 'A' * 30000):
```

Comment:
@juliuszsompolski It looks like Postgres splits long messages into multiple diagnostic records, whereas SQL Server just hands over the whole diagnostic record in one go.  Hence, this test should probably be re-written like so:
```python
        for msg in ('hello world', 'ABCDEFGHIJ' * 400):
            self.cursor.execute("""
                CREATE OR REPLACE PROCEDURE test_cursor_messages()
                LANGUAGE plpgsql
                AS $$
                BEGIN
                    RAISE INFO '{}' USING ERRCODE = '01000';
                END;
                $$;
            """.format(msg))
            self.cursor.execute("CALL test_cursor_messages();")
            messages = self.cursor.messages
            self.assertTrue(type(messages) is list)
            self.assertTrue(len(messages) > 0)
            self.assertTrue(all(type(m) is tuple for m in messages))
            self.assertTrue(all(len(m) == 2 for m in messages))
            self.assertTrue(all(type(m[0]) is unicode for m in messages))
            self.assertTrue(all(type(m[1]) is unicode for m in messages))
            self.assertTrue(all(m[0] == '[01000] (-1)' for m in messages))
            self.assertTrue(''.join(m[1] for m in messages).endswith(msg))
```
Sorry about the code snippet, but it seems the most direct way.

Also, note the "400".  Python 2.7 (64-bit) and/or Postgres doesn't seem to like longer messages.